src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -1043,17 +1060,66 @@ public DynamicObject classNew(VirtualFrame frame, DynamicObject superclass) {
     @CoreMethod(names = "rb_tr_debug", onSingleton = true, rest = true)
     public abstract static class DebugNode extends CoreMethodArrayArgumentsNode {
 
-        @TruffleBoundary
+        @Child CallDispatchHeadNode toSCall;
+
         @Specialization
-        public Object debug(Object... objects) {
+        public Object debug(VirtualFrame frame, Object... objects) {
```

Comment:
Perhaps this method should start with a transfer to interpreter? We don't want to compile this or have this graph in GraalVM do we?

Reply:
Unconditional transfer to interpreter should be forbidden imho.
It would deoptimize everytime when there is no need, we just need a boundary call and find a way to not need that frame (which anyway is likely a Sulong frame and not a Ruby frame so cannot be used to check visibility).